app.py

```diff
@@ -27,19 +46,41 @@ def get(self, restaurant_id):
         for restaurant in restaurants:
             if restaurant["id"] == restaurant_id:
                 return jsonify(restaurant), 200
+        raise CustomError(status_code=404, description="Restaurant with such ID doesn't exist", name="Not Found")
 
     def put(self, restaurant_id):
         content = request.json
+        validate_put(content)
         for restaurant in restaurants:
             if restaurant["id"] == restaurant_id:
                 for key, value in content.items():
                     restaurant[key] = value
                 return jsonify(restaurant), 200
+        raise CustomError(status_code=404,
```

Comment:
and now you have 2 places where you raise `404` error, it can be a sign that it is a time to create `NotFoundError`

Reply:
fixed